src/main/java/com/dsrm/dsrmbackend/services/impl/RoomServiceImpl.java

```diff
@@ -16,19 +18,53 @@
 @Service
 @RequiredArgsConstructor
 public class RoomServiceImpl implements RoomService {
-    private final RoomRepo repository;
+    private final RoomRepo roomRepo;
+    private final RoomTypeRepo roomTypeRepo;
     private final RoomMapper roomMapper;
     public Room addRoom(RoomRequestDTO roomDto) {
         Room room = roomMapper.roomReqDTOToRoom(roomDto);
         room.setId(UUID.randomUUID().toString());
-        return repository.save(room);
+        return roomRepo.save(room);
     }
 
-    public Optional<Room> getRoom(String roleId) {
-        return repository.findById(roleId);
+    public Optional<Room> getRoom(String roomId) {
+        return roomRepo.findById(roomId);
     }
 
     public Page<Room> getRooms(Pageable pageable) {
-        return repository.findAll(pageable);
+        return roomRepo.findAll(pageable);
+    }
+
+    @Override
+    public Room updateRoom(RoomRequestDTO update, String roomId) {
+        Optional<Room> room = roomRepo.findById(roomId);
+        Room newRoom;
+        if (room.isPresent()) {
+            newRoom = room.get();
+            if (update.getNumber() != null) {
+                newRoom.setRoomNumber(update.getNumber());
+            }
+            if (update.getFloor() != null) {
+                newRoom.setFloor(update.getFloor());
+            }
+            Optional<RoomType> roomType = roomTypeRepo.findById(update.getType());
```

Comment:
```roomTypeRepo.getReferenceById(update.getType());```. We don't want to fetch it from database, only put reference.
Look at the userService. And getType can be null so you have to check if it is null before you fetch it. Otherwise exception will be thrown